lib/chef/client.rb

```diff
@@ -676,10 +685,45 @@ def register(client_name = node_name, config = Chef::Config)
     # But is it already there?
     def check_certstore_for_key(cert_name)
       require "win32-certstore"
-      win32certstore = ::Win32::Certstore.open("MY")
+      if Chef::Config[:auth_key_registry_type] == "user"
+        # checking to see if this user is migrating from a cert in LocalMachine store
+        win32certstore = ::Win32::Certstore.open("MY")
+        existing_certificate = win32certstore.search("#{cert_name}")
+        if existing_certificate.nil? || existing_certificate.empty?
+          win32certstore = ::Win32::Certstore.open("MY", store_location: CERT_SYSTEM_STORE_CURRENT_USER)
+        else
+          update_key_and_register(cert_name)
+          delete_old_key(cert_name, CERT_SYSTEM_STORE_CURRENT_USER)
+        end
+      else
+        # checking to see if we're moving a user from the CurrentUser store
+        win32certstore = ::Win32::Certstore.open("MY", store_location: CERT_SYSTEM_STORE_CURRENT_USER)
+        existing_certificate = win32certstore.search("#{cert_name}")
+        if existing_certificate.nil? || existing_certificate.empty?
+          win32certstore = ::Win32::Certstore.open("MY")
+        else
+          update_key_and_register(cert_name)
+          delete_old_key(cert_name, CERT_SYSTEM_STORE_LOCAL_MACHINE)
+        end
+      end
```

Comment:
So we SHOULD migrate old_crypto -> new_crypto.
We should NOT migrate user <-> system.

If `say auth_key_registry_type == :user`, then it should be a separate runlist/nodename/user, and should never touch anything about the system run, and vise-versa.

However, `auth_key_registry_type == :system`, then the key MIGHT be encrypted using conver_to_secure_string, so we need to migrate that _same_ password to encrypted with a vector.

Reply:
done.